python/pyspark/ml/torch/distributor.py

```diff
@@ -1003,3 +1006,142 @@ def _get_spark_partition_data_loader(
         # if num_workers is zero, we cannot set `prefetch_factor` otherwise
         # torch will raise error.
         return DataLoader(dataset, batch_size, num_workers=num_workers)
+
+
+class DeepspeedTorchDistributor(TorchDistributor):
+    
+    def __init__(self, num_gpus: int = 1, nnodes: int = 1, local_mode: bool = True, use_gpu: bool = True, deepspeed_config = None):
+        """
+            This class is used to run deepspeed training workloads with spark clusters. The user has the option to 
+            specify the number of gpus per node and the number of nodes (the same as if running from terminal), 
+            as well as specify a deepspeed configuration file.
+
+            Parameters
+            ----------
+            num_gpus: int
+                the number of GPUs to use per node (analagous to num_gpus in deepspeed command)
+
+            nnodes: int
+                the number of nodes that should be used for the run
+
+            local_mode: bool
+                whether or not to run the training in a distributed fashion or just locally
+
+            use_gpu: bool
+                boolean flag to determine whether to utilize gpus
+
+            deepspeed_config: Union[Dict[str,Any], str] or None:
+                the configuration file to be used for launching the deepspeed application. If it is a dictionary mapping parameters to values, then we will create the file.
+                If None, deepspeed will fall back to default parameters.
+        """
+        num_processes = num_gpus * nnodes
+        super().__init__(num_processes, local_mode, use_gpu)
+        self.deepspeed_config = deepspeed_config 
+        self.ssl_conf = "deepspeed.spark.distributor.ignoreSsl"
+        self._validate_input_params()
+        self.input_params = self._create_input_params()
+        self.cleanup_deepspeed_conf = False
+
+    @staticmethod
+    def _get_deepspeed_config_path(deepspeed_config):
+        if isinstance(deepspeed_config, dict):
+            with tempfile.NamedTemporaryFile(mode='w+', delete=False, suffix='.json') as fil:
+                json.dump(deepspeed_config, fil)
+                return fil.name
+        deepspeed_config_path = deepspeed_config
+        # Empty value means the deepspeed will fall back to default settings.
+        if deepspeed_config == None:
+            deepspeed_config_path = "" 
+
+        return deepspeed_config_path
```

Comment:
Rewrite this as

\# Empty value means the deepspeed will fall back to default settings.
if deepspeed_config == None:
  return ""
return deepspeed_config

Reply:
Done!